src/java/za/co/mip/ablduck/ABLDuckClassVisitor.java

```diff
@@ -0,0 +1,214 @@
+        js.tagname   = "class";
+        js.name      = fullClassName;
+        js.shortname = cu.className;
+        js.classIcon = "class";
+
+        if (cu.inherits != null)
+            js.extends_ = cu.inherits;
+
+        String c = cu.classComment.get(cu.classComment.size() - 1); // Assuming last comment will always be the class comment, will need to cater for license later
+        
+        try {
+            CommentParseResult commentParseResult = CommentParser.parseComment(c, fullClassName);
+
+            //TODO: fix this, not even sure i like it
+            if (!commentParseResult.internal)
+                commentParseResult.internal = cu.className.startsWith("_");
+
+            js.comment = commentParseResult.comment;
+            js.author = commentParseResult.author;
+    
+            if (commentParseResult.internal)
+                js.meta.internal = "This is a private class for internal use by the framework. Don't rely on its existence.";
+    
+    
+            if (!commentParseResult.deprecatedVersion.equals("")) {
+                js.meta.deprecated = new DeprecatedObject();
+    
+                js.meta.deprecated.version = commentParseResult.deprecatedVersion;
+                js.meta.deprecated.text    = commentParseResult.deprecatedText;
+            }
+
+        } catch (IOException ex) {
+            throw ex;
+        }
+
+        Integer methodCount = 0;
+        String previousMethodName = "";
+        // Members
+        for (Method method : cu.methods) {
+            MemberObject m = new MemberObject();
+
+            if (previousMethodName.equals(method.methodName)) {
+                methodCount++;
+            } else {
+                methodCount = 0;
+                previousMethodName = method.methodName;
+            }
+
+            if (methodCount == 0)
+                m.id = "method-" + method.methodName;
+            else
+                m.id = "method-" + method.methodName + "-" + methodCount.toString();
+            m.name = method.methodName;
+            m.owner = fullClassName;
+            m.tagname = "method";
+            m.signature = method.signature;
+
+            try {
+                CommentParseResult commentParseResult = CommentParser.parseComment(method.methodComment, fullClassName + ":" + method.methodName);
+
+                if (!commentParseResult.internal)
+                    commentParseResult.internal = method.methodName.startsWith("_");
+
+                m.returnComment = commentParseResult.returnComment;
+                m.comment = commentParseResult.comment;
+
+                if (commentParseResult.internal)
+                    m.meta.internal = "This is a private method for internal use by the framework. Don't rely on its existence.";
+
+                //TODO: remove this duplication tsk tsk, write a meta adding method for class, method and property
+                if (!commentParseResult.deprecatedVersion.equals("")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)